src/View/AbstractTemplateView.php

```diff
@@ -221,6 +221,9 @@ public function renderSection($sectionName, array $variables = [], $ignoreUnknow
         if ($this->getCurrentRenderingType() === self::RENDERING_LAYOUT) {
             // in case we render a layout right now, we will render a section inside a TEMPLATE.
             $renderingTypeOnNextLevel = self::RENDERING_TEMPLATE;
+            foreach ($variables as $key => $value) {
```

Comment:
This is the dangerous line. It operates on the not-cloned RenderingContext retrieved from `getCurrentRenderingContext` so assigning variables here, bleeds the variables to other sections and the layout's subsequent code.

Reply:
```
        $renderingContext = $this->getCurrentRenderingContext();
        $renderingTypeOnNextLevel = $this->getCurrentRenderingType();
        if ($this->getCurrentRenderingType() === self::RENDERING_LAYOUT) {
            // in case we render a layout right now, we will render a section inside a TEMPLATE.
            $renderingTypeOnNextLevel = self::RENDERING_TEMPLATE;
            if (empty($variables)) {
                $variables = $renderingContext->getVariableProvider()->getAll();
            }
        }
        $variableProvider->getVariableProvider()->getScopeCopy($variables)
        $renderingContext = clone $renderingContext;
        $renderingContext->setVariableProvider($variableProvider);
```